contracts/script/deploy/DeployT1ERC7683Pull.s.sol

```diff
@@ -0,0 +1,103 @@
+        address L1_T1_X_CHAIN_READ_PROXY_ADDR = vm.envAddress("L1_T1_X_CHAIN_READ_PROXY_ADDR");
+        address L1_PROXY_ADMIN_ADDR = vm.envAddress("L1_PROXY_ADMIN_ADDR");
+        proxyAdmin = ProxyAdmin(L1_PROXY_ADMIN_ADDR);
+
+        vm.startBroadcast(deployerPk);
+
+        // Deploy L1 router implementation
+        T1ERC7683Pull impl = new T1ERC7683Pull(
+            L1_T1_MESSENGER_PROXY_ADDR,
+            address(0), // No Permit2 for now
+            L1_T1_X_CHAIN_READ_PROXY_ADDR,
+            PR1
+        );
+
+        // Deploy and initialize proxy
+        TransparentUpgradeableProxy proxy =
+            new TransparentUpgradeableProxy(address(impl), address(proxyAdmin), new bytes(0));
+
+        vm.stopBroadcast();
+
+        logAddress("L1_T1_PULL_BASED_7683_IMPLEMENTATION_ADDR", address(impl));
+        logAddress("L1_T1_PULL_BASED_7683_PROXY_ADDR", address(proxy));
+
+        logEnd("DeployRouterPullBasedERC7683 to PR1");
+    }
+
+    function pr1_init() external {
+        uint256 deployerPk = vm.envUint("L1_DEPLOYER_PRIVATE_KEY");
+        address L1_T1_7683_PROXY_ADDR = vm.envAddress("L1_T1_PULL_BASED_7683_PROXY_ADDR");
+        address L2_T1_7683_PROXY_ADDR = vm.envAddress("L2_T1_PULL_BASED_7683_PROXY_ADDR");
+
+        vm.startBroadcast(deployerPk);
+
+        T1ERC7683Pull(L1_T1_7683_PROXY_ADDR).initialize(L2_T1_7683_PROXY_ADDR);
+
+        vm.stopBroadcast();
+    }
+
+    function t1_deploy() external {
+        logStart("DeployRouterPullBasedERC7683 to t1");
+        vm.createSelectFork(vm.rpcUrl("t1"));
+        uint256 deployerPk = vm.envUint("L2_DEPLOYER_PRIVATE_KEY");
+        address L2_T1_MESSENGER_PROXY_ADDR = vm.envAddress("L2_T1_MESSENGER_PROXY_ADDR");
+        address L2_T1_X_CHAIN_READ_PROXY_ADDR = vm.envAddress("L2_T1_X_CHAIN_READ_PROXY_ADDR");
+        address L2_PROXY_ADMIN_ADDR = vm.envAddress("L2_PROXY_ADMIN_ADDR");
+
+        vm.startBroadcast(deployerPk);
+
+        proxyAdmin = ProxyAdmin(L2_PROXY_ADMIN_ADDR);
+
+        // Deploy L2 router implementation
+        T1ERC7683Pull impl = new T1ERC7683Pull(
+            L2_T1_MESSENGER_PROXY_ADDR,
+            address(0), // No Permit2 for now
+            L2_T1_X_CHAIN_READ_PROXY_ADDR,
+            T1
+        );
+
+        // Deploy and initialize proxy
+        TransparentUpgradeableProxy proxy =
+            new TransparentUpgradeableProxy(address(impl), address(proxyAdmin), new bytes(0));
+
+        vm.stopBroadcast();
+
+        logAddress("L2_T1_PULL_BASED_7683_IMPLEMENTATION_ADDR", address(impl));
+        logAddress("L2_T1_PULL_BASED_7683_PROXY_ADDR", address(proxy));
+
+        logEnd("DeployRouterPullBasedERC7683 to t1");
+    }
+
+    function t1_init() external {
+        vm.createSelectFork(vm.rpcUrl("t1"));
+        uint256 deployerPk = vm.envUint("L2_DEPLOYER_PRIVATE_KEY");
+        address L1_T1_7683_PROXY_ADDR = vm.envAddress("L1_T1_PULL_BASED_7683_PROXY_ADDR");
```

Comment:
This var and the following are missing in the `.env.template` file.

Reply:
added